src/main/java/com/iota/iri/controllers/LocalSnapshotViewModel.java

```diff
@@ -0,0 +1,129 @@
+package com.iota.iri.controllers;
+
+import com.iota.iri.model.Hash;
+import com.iota.iri.model.LocalSnapshot;
+import com.iota.iri.storage.PersistenceProvider;
+
+import java.util.Map;
+
+/**
+ * Acts as a controller interface for a {@link LocalSnapshot}.
+ */
+public class LocalSnapshotViewModel {
```

Comment:
Since DB refactoring didn't happen yet we have `ViewModels` in our code.
So you went ahead and also created ones, which is the reasonable thing to do.

The problem now is that local snapshot logic was written in Domain Driven Design style.
It was supposed to be an example to how the rest of the code should look after refactoring.

So now we have `LocalSnapshot` persistable,`LocalSnapshotViewModel`, `SnapshotProvider`, `SnapshotService`, `Snapshot`, and `SnapshotMetaData`.

I am pretty sure that you also find this confusing.

So in the new design there should be only:
1 . "Model": `Snapshot` that implements `Persistable`. Technically in this design, `Persistable` should be a marker interface and not have any of the current methods.  Maybe only `merge` (not sure atm). However, because of how `RocksDBPersistenceProvider` is implemented this is currently a problem. 
2. "Provider"/"Repository" - `SnapshotProvider` - a singleton that should be in charge of marshalling / unmarshalling the snapshot data. It also caches `initial` and `latest` snapshot.
3. "Service" - `SnapshotService` - In charge of business logic and performing calculations that require more dependencies.

Anyhows, if we were using the old design there would be:
1. Model `LocalSnapshot` - The `Persistable` representation for the DB
2. ViewModel `LocalSnapshotViewModel` -  In the context of IRI those view models are in practice all the concerns I mentioned before for DDD mashed up in one evil class.


Anyhows, since currently IRI has those 2 design paradigms (since we didn't consolidate them into one yet), it caused confusion that lead to this combination of the 2 paradigms.

Honestly, since `RocksDBPersisitenceProvider` needs to be changed in order to use the DDD paradigm I am not sure whether you should change anything yet... I suppose maybe you can add some of the Provider functionality into the the `Snapshot` model (`bytes` and `read`). Then it will be only somewhat ugly, but maybe we can erase `LocalSnapshotViewModel` and `LocalSnapshot`...

On the monday meeting we can discuss this. Maybe it can be done as a separate issue.